pyam/ixmp4.py

```diff
@@ -114,7 +123,15 @@ def write_to_ixmp4(platform: ixmp4.Platform | str, df):
         _df = df.filter(model=model, scenario=scenario)
 
         run = platform.runs.create(model=model, scenario=scenario)
-        run.iamc.add(_df.data)
+
+        if df.time_domain == "year":
+            run.iamc.add(_df.data)
+        else:
```

Comment:
I know this is technically not required since we only reach this point if we have datetime or year anyway but I always prefer explicit code:
```suggestion
        elif df.time_domain == "datetime":
```
since the code inside the if clause will only work for datetime